cray/modules/ims/openapi.yaml

```diff
@@ -26,131 +26,117 @@ openapi: 3.0.2
-      and then an interactive ssh shell
-      in which to modify the copy of the image.  Upon success,
-      the artifact repository contains the original image and a modified version of it. IMS contains 
-      a new ImageRecord - metadata for the modified image.
-      The original image is still intact.  A user may want to install additional software, 
-      install licenses, change the timezone, add mount points, etc.
-    
-  version: 3.1.0
+      There are two main workflows using the IMS - image creation and image customization.
+      The IMS /jobs endpoint directs the creation of a new image, or the customization of an
+      existing image, depending on the POST /jobs request job_type body parameter.
+
+      ### Add a New Recipe
+
+        #### GET /recipes
+
+          Obtain list of existing recipes which are registered with IMS.
+
+        #### Upload recipe using CLI
+
+          Upload a new recipe to the artifact repository using the cray artifacts command, if necessary.
+          Refer to Administrator's Guide for instructions.
+
+        #### POST /recipes
+
+          Register new recipe with IMS.
+
+      ### Manage Public Keys
+
+        #### GET /public-keys
+
+          Obtain list of available public-keys.
+
+        #### POST /public-keys
+
+          Add a new public-key.
+
+      ### Create a New Image
+
+        #### GET /public-keys
+
+          Get a list of available keys.
+
+        #### GET /recipes
+
+          Get recipe ID.
+
+        #### POST /jobs
+
+          Use Kiwi-NG to create a new IMS image and image artifacts from a recipe. Specify job_type
+          "create" in JobRecord. Request body parameters supply the recipe ID and public key ID.
+          Upon success, the artifact repository contains the new image and the image artifacts,
+          IMS contains a new ImageRecord with metadata for the new image. During the creation
+          process, IMS may create an SSH shell for administrator interaction with the image for
+          debugging, if necessary.  (enable_debug = true in JobRecord)
+
+      ### Modify an Image
+
+        #### GET /public-keys
+
+          Get a list of available keys.
+
+        #### GET /images
+
+          Obtain a list of available images registered in IMS.
+
+        #### POST /jobs
+
+          To create a modified version of an existing IMS image, specify job_type "customize".
+          Specify the IMS ID of the existing image, and public key.  This request creates a copy of
+          the existing image, and then an interactive SSH shell in which to modify the copy of the
+          image. Upon success, the artifact repository contains the original image and a modified
+          version of it. IMS contains a new ImageRecord with metadata for the modified image. The
+          original image is still intact.  A user may want to install additional software, install
+          licenses, change the timezone, add mount points, etc.
+
+  version: "0.0.0-imsserv"
```

Comment:
cms-meta-tools will replace this version?

Reply:
Not here in the craycli repository. Once [the source PR](https://github.com/Cray-HPE/ims/pull/62) merges, though, I can update this PR with the actual version number. I just didn't want to do it yet, since I didn't know for sure what the IMS version would end up being.

Reply:
> cms-meta-tools will replace this version?

@jsollom-hpe Not here in the craycli repository. Once [the source PR](https://github.com/Cray-HPE/ims/pull/62) merges, though, I can update this PR with the actual version number. I just didn't want to do it yet, since I didn't know for sure what the IMS version would end up being.

Reply:
Although I also don't think that version number matters -- the only version the CLI looks at (AFAIK) is the API version as it relates to endpoints (v1, v2, etc) -- I don't think it uses this version field.